packages/core/adder/preconditions.ts

```diff
@@ -40,6 +43,29 @@ function getGlobalPreconditions(
                     }
                 },
             },
+            {
+                name: "supported environments",
+                // eslint-disable-next-line @typescript-eslint/require-await
```

Comment:
we can get rid of the need for this line by tweaking the type definition of `run`. rather than using a `Promise`, we can use a `MaybePromise`:
```ts
type MaybePromise<T> = Promise<T> | T;

// core/adder/config.ts
 export type Precondition = {
    name: string;
    run: () => MaybePromise<{ success: boolean; message: string | undefined }>;
};
```
we wouldn't have to change anything else either, since `await` at the invocation sites can work for both an `async` and regular function

Reply:
JS is so fancy sometimes...
Good catch, done.